### PR TITLE
migrate metric learning example to keras 3

### PR DESCRIPTION
**Pull request by Shi-pra-19**
Refactor loss function and tensor handling in metric learning for compatibility with Keras 3.

### CODE REVIEW

**Comment by Shi-pra-19** [comment]
I added `run_eagerly=True` because, without it, the training raised an **"OperatorNotAllowedInGraphError"** caused by an internal operation in the TensorFlow Similarity library.
